### DISCUSSION TITLE
How to use Built-in Cache attributes?

### DISCUSSION THREAD

**Discussion by stephanvierkant** [symfony/symfony#48375] [Q&A]
I'm trying to switch from the [SensioFrameworkExtraBundle](https://symfony.com/bundles/SensioFrameworkExtraBundle/current/index.html) to the new built-in cache attributes from Symfony 6.2, but I'm getting an error.

```diff
- #[\Sensio\Bundle\FrameworkExtraBundle\Configuration\Cache(maxage: 120, lastModified: 'entity.getUpdatedAt()', etag: 'entity.getETag()')]
+ #[\Symfony\Component\HttpKernel\Attribute\Cache(maxage: 120, lastModified: 'entity.getUpdatedAt()', etag: 'entity.getETag()')]
public function someAction(Entity $entity) : Response
{
}
```

This used to work, but now it gives an error:

> Variable "entity" is not valid around position 1 for expression `entity.getUpdatedAt()`.

Any idea why this doesn't work? I can dive into it, but maybe someone has already found out why this doesn't work.

**Comment by HypeMC** [comment]
This looks like a bug. The `CacheAttributeListener` listens for a `kernel.controller` event, but the `EntityValueResolver` resolves the entity afterwards:
https://github.com/symfony/symfony/blob/97f0ce875d87ef566bf533e34fc8419e36f84e6f/src/Symfony/Component/HttpKernel/HttpKernel.php#L150-L155

You should open a bug report.

**Reply by stephanvierkant** [reply]
Done! https://github.com/symfony/symfony/issues/48506
